datadog_checks_base/datadog_checks/base/utils/http.py

```diff
@@ -318,7 +318,10 @@ def populate_options(self, options):
 
         for option, value in iteritems(self.options):
             # Make explicitly set options take precedence
-            options.setdefault(option, value)
+            if option == 'headers' and 'headers' in option:
```

Comment:
```suggestion
            if option == 'headers' and 'headers' in options:
```
?

Reply:
Yes, of course. Spotted that when adding the tests.